src/logging/log_dest_file.cpp

```diff
@@ -136,6 +139,10 @@ void file_log_disable() {
     log_info(FileSystem, "Stopped logging to file");
 
     std::scoped_lock guard(mutex);
+    while (!data->buffer.isEmpty()) {
```

Comment:
A better place for this cleanup logic would be in `~Data()`. Don't forget to check for the file not being null

Reply:
You forgot to check the file for being null in the destructor